File: src/Servers/QueryReport/src/V2/Abstraction/BaseClass/RequestBase.cs
using System;
using System.Linq;
using UniSpy.Server.QueryReport.V2.Enumerate;
using UniSpy.Server.QueryReport.Exception;

namespace UniSpy.Server.QueryReport.V2.Abstraction.BaseClass
{
    public abstract class RequestBase : UniSpy.Server.Core.Abstraction.BaseClass.RequestBase
    {
        public static readonly byte[] MagicData = { 0xFE, 0XFD };
        public uint? InstantKey { get; protected set; }
        public new RequestType CommandName { get => (RequestType)base.CommandName; set => base.CommandName = value; }
        public new byte[] RawRequest { get => (byte[])base.RawRequest; protected set => base.RawRequest = value; }

        public RequestBase(byte[] rawRequest) : base(rawRequest)
        {
        }

        protected RequestBase()
        {
        }

        public override void Parse()
        {
            if (RawRequest.Length < 3)
            {
                throw new QRException("Query report request is invalid.");
            }
            CommandName = (RequestType)RawRequest[0];
            var instantKeyBytes = RawRequest.Skip(1).Take(4).ToArray();
            InstantKey = BitConverter.ToUInt32(instantKeyBytes);
        }
    }
}

File: src/servers/query_report/v2/abstractions/request_base.py
import library.abstractions.contracts
from servers.query_report.exceptions.exceptions import QRException
from servers.query_report.v2.enums.general import RequestType

MAGIC_DATA = [0xFE, 0xFD]


class RequestBase(library.abstractions.contracts.RequestBase):
    instant_key: int
    command_name: RequestType
    raw_request: bytes

    def __init__(self, raw_request: bytes) -> None:
        assert isinstance(raw_request, bytes)
        super().__init__(raw_request)

    def parse(self):
        if len(self.raw_request) < 3:
            raise QRException
        self.command_name = RequestType(self.raw_request[0])
        self.instant_key = int(self.raw_request[1:5])
